letter-shop.md

```diff
@@ -0,0 +1,190 @@
+        triggered_at: new Date().toISOString(),
+        forwarded_by: [],
+        additional_info: {}
+      })
+    } else {
+      // the ledger will check if the fulfillment is correct and if it was submitted before the transfer's
+      // rollback timeout
+      plugin.fulfillCondition(transfer.id, fulfillments[transfer.executionCondition]).catch(function () {})
+    }
+  })
+
+  // [...]
+
+})
+```
+
+Since Interledger connects potentially very different ledgers together, we need an abstraction layer that hides the specifics of the ledger, but exposes the interface needed to send and receive money over that ledger. That is what Interledger plugins are for. The 'XRP escrow' plugin is a wrapper around [RippleLib](https://github.com/ripple/ripple-lib), and exposes the Ledger Plugin Interface (LPI). In the code, you see that an 'ilp-plugin-xrp-escrow' Plugin is being configured with secret, account, server, and prefix. The first three come from the [XRP Testnet Faucet](https://ripple.com/build/xrp-test-net/).
+
+### Interledger addresses
+
+The prefix is an Interledger prefix, which is like an [IP subnet](https://en.wikipedia.org/wiki/Subnetwork). In this case, `test.` indicates that we are connecting to the Interledger testnet-of-testnets. The next part, `crypto.` indicates that we will be referring to a crypto currency's ledger. And finally, `xrp.` indicates that this ledger is the XRP testnet ledger. If you know the ledger prefix and the account, you can put them together to get the Interledger Address (see [IL-RFC-15, draft 1](https://interledger.org/rfcs/0015-ilp-addresses/draft-1.html)). In this case, the Interledger address of our Letter Shop is `test.crypto.xrp.rrhnXcox5bEmZfJCHzPxajUtwdt772zrCW`.
+
+### Prepare and Fulfill with on-ledger Escrow
+```js
+function base64url (buf) { return buf.toString('base64').replace(/\+/g, '-').replace(/\//g, '_').replace(/=+$/, '') }
+
+// [...]
+
+const secret = crypto.randomBytes(32)
+const fulfillment = base64url(secret)
+const condition = base64url(crypto.createHash('sha256').update(secret).digest())
+```
+
+The plugin used here is specific to XRP, and to the use of on-ledger escrow. Escrow for XRP is described [here](https://ripple.com/build/rippleapi/#transaction-types). Escrow transfers differ from normal transfer in that the recipient doesn't automatically receive the amount of the transfer in their account; the need to produce something in order to claim the funds. During the time between the sender's action of preparing the transfer (creating the escrow), and the time the recipient produces the fulfillment for the transfer's condition, the money is on hold on the ledger. If the recipient doesn't produce the fulfillment in time, the transaction is canceled, and the money goes back into the sender's account.
+
+### Hashlocks
+In the case of Interledger, the fulfillment is always a 32-byte string, and the condition is the sha256 hash of that string.
```

Comment:
This description kind of reads like you expect the reader to know what condition and fulfillment mean, but those terms are somewhat specific to Interledger